Hive SSO guides for Okta and Azure AD (#1017)
kamilkisiela/graphql-hive: packages/web/docs/src/pages/features/sso-oidc-provider.mdx

type", then click "Next".

![](/docs/pages/features/sso-oidc-provider/create-new-app-integration-options.png)

Next you're gonna be able to customize your app's name and logo if you want to, then choose the
option that suits you for the "Controlled access", I'm gonna go with "Allow everyone" as this's just
a guide, after you're done click "Save".

![](/docs/pages/features/sso-oidc-provider/controlled-access-of-okta-app.png)

And now you're almost good to go with Okta's configuration, now head to your Hive organization
settings tab and hit the "Connect Open ID Connect Provider" button

![](/docs/pages/features/sso-oidc-provider/hive-connect-openid-provider.png)

And now you'd have to fill in a couple of inputs to connect your OpenID Okta's provider as the
following:

![](/docs/pages/features/sso-oidc-provider/hive-connect-openid-provider-modal.png)

- Token Endpoint = https://${yourOktaDomain}/oauth2/v1/token (ex:
  https://trial-5517233.okta.com/oauth2/v1/token)
- User Info Endpoint = https://${yourOktaDomain}/oauth2/v1/userinfo (ex:
  https://trial-5517233.okta.com/oauth2/v1/userinfo)

> - Authorization Endpoint = https://${yourOktaDomain}/oauth2/v1/authorize (ex:

Author: ```suggestion
- Token Endpoint = https://YOUR_OKTA_DOMAIN_HERE/oauth2/v1/token (ex:
  https://trial-5517233.okta.com/oauth2/v1/token)
- User Info Endpoint = https://YOUR_OKTA_DOMAIN_HERE//oauth2/v1/userinfo (ex:
  https://trial-5517233.okta.com/oauth2/v1/userinfo)
- Authorization Endpoint = https://YOUR_OKTA_DOMAIN_HERE//oauth2/v1/authorize (ex:
```

  https://trial-5517233.okta.com/oauth2/v1/authorize)
- Client ID = ![](/docs/pages/features/sso-oidc-provider/okta-app-config-client-id.png)
- Client Secret = ![](/docs/pages/features/sso-oidc-provider/okta-app-config-client-secret.png)

After you're done, click the "Connect OIDC Provider" button.

The last step to do is, grabbing both the OIDC provider sign-in and sign-out redirect URIs by
clicking the copy button next to them, then go back to your Okta's app SSO configuration and scroll
down to the "General Settings" section and click "Edit" on the right, then scroll down to the
"LOGIN" section and paste both the Sign-in redirect URI and the Sign-out redirect URI from Hive into
their responding inputs and click "Save" when you're done.

![](/docs/pages/features/sso-oidc-provider/hive-manage-openid-connect-modal.png)

![](/docs/pages/features/sso-oidc-provider/okta-app-general-settings-section.png)

> And congrats, you know have Okta connected! 🥳🎉

Author: ```suggestion
And congrats, you now have Okta connected! 🥳🎉
```

![](/docs/pages/features/sso-oidc-provider/azure-ad-configure-api-permissions.png)
![](/docs/pages/features/sso-oidc-provider/azure-ad-configure-api-permissions-graph.png)
![](/docs/pages/features/sso-oidc-provider/azure-ad-delegated-permissions.png)

An Important Note:

<Callout type="info">
  Granting the permission for `email` unfortunately doesn't guarantee a successful login, because
  how Azure AD works is that emails are optional which is problematic for Hive as an email is what
  makes a login successful, so make sure when adding user records to add their email field as well
  so that they're able to login. Azure made it extra annoying because the only way to add an email
  is by editing an existing user not when creating a user record, but it is what it is. You can
  follow the steps in the screenshots below to be able to add an email for a user record.
  ![](/docs/pages/features/sso-oidc-provider/azure-ad-users-tab.png)
  ![](/docs/pages/features/sso-oidc-provider/azure-ad-users-table.png)
  ![](/docs/pages/features/sso-oidc-provider/azure-ad-edit-user-properties.png)
  ![](/docs/pages/features/sso-oidc-provider/azure-ad-edit-user-email-field.png)
</Callout>

> And congrats, you know have Azure AD connected! 🥳🎉

Author: ```suggestion
And congrats, you now have Azure AD connected! 🥳🎉
```

people will be redirected to it.

E.g. `https://app.graphql-hive.com/auth/oidc?id=xxxxxxxx-1234-1234-1234-xxxxxxxxxxxx`

## Configuring with Okta

> First we need to have an Okta account, so you can register for an account from

Reviewer: I think we can skip the trial part. If somebody is interested in connecting Okta to Hive, they probably have Okta account already.

Author: Yeah, agree

https://login.okta.com/signin/register or sign up for a 30 days trial from
https://www.okta.com/free-trial to ease the process of going through this guide.

> Note: that unfortunately for sigining up for an Okta account, you're required to have a busieness

> > email.

Reviewer: It's the requirement of Okta, we should skip it in our docs.

Author: ok

> First we need to have an Okta account, so you can register for an account from
> https://login.okta.com/signin/register or sign up for a 30 days trial from
> https://www.okta.com/free-trial to ease the process of going through this guide.
> 
> > Note: that unfortunately for sigining up for an Okta account, you're required to have a busieness
> > email.
> 

Author: ```suggestion
```

- Client Secret = ![](/docs/pages/features/sso-oidc-provider/okta-app-config-client-secret.png)

After you're done, click the "Connect OIDC Provider" button.

The last step to do is, grabbing both the OIDC provider sign-in and sign-out redirect URIs by
clicking the copy button next to them, then go back to your Okta's app SSO configuration and scroll
down to the "General Settings" section and click "Edit" on the right, then scroll down to the
"LOGIN" section and paste both the Sign-in redirect URI and the Sign-out redirect URI from Hive into
their responding inputs and click "Save" when you're done.

![](/docs/pages/features/sso-oidc-provider/hive-manage-openid-connect-modal.png)

![](/docs/pages/features/sso-oidc-provider/okta-app-general-settings-section.png)

And congrats, you know have Okta connected! 🥳🎉

You can use the link provided in the modal right section to let your users login via Okta.

![](/docs/pages/features/sso-oidc-provider/hive-manage-openid-okta-login-url.png)

Hope that helped, and let us know if something changed on either Okta's or Hive's side in the
screenshots and/or steps and we'll make that change happen.

## Configuring with Azure AD

> First we need to have a Microsoft Azure account, so you can register for a free account from

Reviewer: same as with Okta

> First we need to have a Microsoft Azure account, so you can register for a free account from
> https://azure.microsoft.com/en-us/free

Author: ```suggestion
```

![](/docs/pages/features/sso-oidc-provider/azure-ad-configure-api-permissions.png)
![](/docs/pages/features/sso-oidc-provider/azure-ad-configure-api-permissions-graph.png)
![](/docs/pages/features/sso-oidc-provider/azure-ad-delegated-permissions.png)

An Important Note:

<Callout type="info">
  Granting the permission for `email` unfortunately doesn't guarantee a successful login, because
  how Azure AD works is that emails are optional which is problematic for Hive as an email is what
  makes a login successful, so make sure when adding user records to add their email field as well
  so that they're able to login. Azure made it extra annoying because the only way to add an email
  is by editing an existing user not when creating a user record, but it is what it is. You can
  follow the steps in the screenshots below to be able to add an email for a user record.
  ![](/docs/pages/features/sso-oidc-provider/azure-ad-users-tab.png)
  ![](/docs/pages/features/sso-oidc-provider/azure-ad-users-table.png)
  ![](/docs/pages/features/sso-oidc-provider/azure-ad-edit-user-properties.png)
  ![](/docs/pages/features/sso-oidc-provider/azure-ad-edit-user-email-field.png)
</Callout>

> And...Woohoo, you now have Azure AD connected! 🥳🎊

Author: ```suggestion
And congrats, you know have Azure AD connected! 🥳🎉
```
